packages/api/src/derives/fees/estimateFee.ts

```diff
@@ -0,0 +1,66 @@
+import {ApiInterfaceRx} from '@cennznet/api/types';
+import Extrinsic from '@cennznet/types/extrinsic/Extrinsic';
+import {ChargeTransactionPayment} from '@cennznet/types/runtime/transaction-payment';
+import {AnyAssetId, IExtrinsic} from '@cennznet/types/types';
+import {drr} from '@polkadot/rpc-core/rxjs';
+import {TypeRegistry} from '@polkadot/types';
+import {Address} from '@polkadot/types/interfaces';
+import {RuntimeDispatchInfo} from '@polkadot/types/interfaces/rpc';
+import ExtrinsicEra from '@polkadot/types/primitive/Extrinsic/ExtrinsicEra';
+import BN from 'bn.js';
+import {combineLatest, Observable, of} from 'rxjs';
+import {catchError, first, map, switchMap} from 'rxjs/operators';
+
+export function estimateFee(api: ApiInterfaceRx) {
+  /* To estimate fee correctly using extrinsic's fake signature */
```

Comment:
Something more descriptive would be good e.g.
// We generate fake signature data here to ensure the estimated fee will correctly match the fee paid when the extrinsic is signed by a user.
// This is because fees are currently based on the byte length of the extrinsic

Reply:
added exactly above comments